includes/classes/class-fixinsecurecontent.php

```diff
@@ -0,0 +1,317 @@
+<?php
+/**
+ * Handles the fixing of insecure content in posts for REST and WP CLI requests.
+ *
+ * @package ICW
+ */
+
+namespace ICW\FIX;
+
+use WP_CLI;
+use WP_CLI\Utils;
+use WP_CLI_Command;
+use WP_Query;
+
+/**
+ * FixInsecureContent class.
+ */
+class FixInsecureContent {
+	/**
+	 * Is current run a dry run?
+	 *
+	 * @var bool
+	 */
+	private bool $dry_run;
+
+	/**
+	 * Total number of posts scanned for URLs.
+	 *
+	 * @var int
+	 */
+	private int $total_post_count = 1;
+
+	/**
+	 * Data of posts where 1 or more URLs were found to be insecure and command fixed summary.
+	 *
+	 * @var array
+	 */
+	private array $fixed_post_count;
+
+	/**
+	 * Array of warning messages.
+	 *
+	 * @var array
+	 */
+	private array $warning_count = array();
+
+	/**
+	 * Class instance
+	 *
+	 * @var ?FixInsecureContent
+	 */
+	protected static ?FixInsecureContent $instance = null;
+
+	/**
+	 * Get class instance
+	 *
+	 * @return FixInsecureContent
+	 */
+	public static function get_instance() {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Counts posts to be checked for insecure content.
+	 *
+	 * @param string $post_type      The post type.
+	 * @param int    $posts_per_page The batch size.
+	 * @param int    $post_offset    The post offset.
+	 *
+	 * @return int
+	 */
+	public function count_post_to_check( string $post_type, int $posts_per_page, int $post_offset ): int {
+		$args = array(
+			'posts_per_page' => $posts_per_page,
+			'post_type'      => 'all' === $post_type ? 'any' : $post_type,
```

Comment:
This should be limited to only posts available via the rest api -- per note re:schema. Many e-commerce plugins use post types for orders sow we definitely need to exclude those.

Reply:
> The WP CLI functionality can properly scan/fix post types that are not shown in REST, so in my opinion, the admin functionality should behave exactly the same.

The difference between the two is that the WP-CLI is basically the equivalent of running as the `root` user on a server. Once someone has access to the server they can do anything, such as running `wp db clean --yes` so permissions don't apply.

For users with access to the dashboard only, we need to check that they have full permission to make any changes to a post. If a post type isn't editable via the rest API then we need to assume the developer has reasons for preventing this and respect their decision

Reply:
What you say makes perfect sense @peterwilsoncc. I've modified the method that counts posts with and the method that fixes insecure content so that posts from post types that are not shown in RFEST are excluded from the process.

Please let me know if this is what you meant. Thank you for your feedback and patience.